drivers/l3gd20h/l3gd20h.c

```diff
@@ -0,0 +1,991 @@
+        if (!_get_reg_bit(reg1, L3GD20H_POWER_MODE)) {
+            xtimer_usleep(200000);
+        }
+
+        if (mode <= L3GD20H_MODE_12) {
+            /* high output data rate */
+            _set_reg_bit(&reg2, L3GD20H_LOW_ODR, 0);
+            _set_reg_bit(&reg1, L3GD20H_ODR_BW, mode);
+        }
+        else {
+            /* low output data rate */
+            _set_reg_bit(&reg2, L3GD20H_LOW_ODR, 1);
+            _set_reg_bit(&reg1, L3GD20H_ODR_BW, mode - L3GD20H_MODE_12);
+        }
+        _set_reg_bit(&reg1, L3GD20H_POWER_MODE, 1);
+        _set_reg_bit(&reg1, L3GD20H_X_ENABLED, x);
+        _set_reg_bit(&reg1, L3GD20H_Y_ENABLED, y);
+        _set_reg_bit(&reg1, L3GD20H_Z_ENABLED, z);
+
+        if (dev->sensor == L3GD20H) {
+            EXEC_RET(l3gd20h_reg_write(dev, L3GD20H_REG_LOW_ODR, &reg2, 1), res);
+        }
+    }
+    else {
+        _set_reg_bit(&reg1, L3GD20H_POWER_MODE, 0);
+    }
+
+    EXEC_RET(l3gd20h_reg_write(dev, L3GD20H_REG_CTRL1, &reg1, 1), res);
+
+    dev->params.mode = mode;
+
+    return res;
+}
+
+int l3gd20h_set_scale(l3gd20h_t *dev, l3gd20h_scale_t scale)
+{
+    ASSERT_PARAM(dev != NULL);
+    DEBUG_DEV("scale=%02x", dev, scale);
+
+    dev->params.scale = scale;
+
+    /* read CTRL4 register and write scale */
+    return _update_reg(dev, L3GD20H_REG_CTRL4, L3GD20H_FULL_SCALE, scale);
+}
+
+int l3gd20h_power_down (l3gd20h_t *dev)
+{
+    ASSERT_PARAM(dev != NULL);
+    DEBUG_DEV("", dev);
+
+    return _update_reg(dev, L3GD20H_REG_CTRL1, L3GD20H_POWER_MODE, 0);
+}
+
+int l3gd20h_power_up (l3gd20h_t *dev)
+{
+    ASSERT_PARAM(dev != NULL);
+    DEBUG_DEV("", dev);
+
+    int res = _update_reg(dev, L3GD20H_REG_CTRL1, L3GD20H_POWER_MODE, 1);
+
+    /* it takes up to 100 ms to switch from power down to another mode */
+    xtimer_usleep(100000);
+
+    return res;
+}
+
+int l3gd20h_sleep (l3gd20h_t *dev)
+{
+    ASSERT_PARAM(dev != NULL);
+    DEBUG_DEV("", dev);
+
+    return _update_reg(dev, L3GD20H_REG_CTRL1, L3GD20H_XYZ_ENABLED, 0);
+}
+
+int l3gd20h_wake_up (l3gd20h_t *dev)
+{
+    ASSERT_PARAM(dev != NULL);
+    DEBUG_DEV("", dev);
+
+    int res = _update_reg(dev, L3GD20H_REG_CTRL1, L3GD20H_XYZ_ENABLED, 7);
+
+    return res;
+}
+
+int l3gd20h_data_ready(const l3gd20h_t *dev)
+{
+    ASSERT_PARAM(dev != NULL);
+    DEBUG_DEV("", dev);
+
+    int res = L3GD20H_OK;
+
+    uint8_t reg;
+
+    if (dev->params.fifo_mode == L3GD20H_BYPASS) {
+        EXEC_RET(l3gd20h_reg_read(dev, L3GD20H_REG_STATUS, &reg, 1), res);
+        return _get_reg_bit(reg, L3GD20H_ANY_DATA_READY);
+    }
+    else {
+        EXEC_RET(l3gd20h_reg_read(dev, L3GD20H_REG_FIFO_SRC, &reg, 1), res);
+        return _get_reg_bit(reg, L3GD20H_FIFO_FFS);
+    }
+}
+
+/*
+ * scale factors for conversion of raw sensor data to degree for possible
+ * sensitivities according to mechanical characteristics in datasheet
+ */
+static const float L3GD20H_SCALES[3] = { (0.00875F), (0.0175F), (0.070F) };
```

Comment:
```suggestion
static const uint16_t L3GD20H_SCALES[3] = { 35, 70, 280 };
```

multiplied everything by 4000